### PR TITLE
Jmprieur/separate graph

### PR DESCRIPTION
**Pull request by jmprieur**
Fix for #506:

- Moves the classes which were under the MicrosoftGraph folder into a new assembly:**Microsoft.Identity.Web.MicrosoftGraph**
- Updated the samples calling Graph
- Updated the templates to conditionnaly reference *Microsoft.Identity.Web.MicrosoftGraph**
- Also adds **Microsoft.Identity.Web.MicrosoftGraphBeta**

### CODE REVIEW

**Comment by jmprieur** [comment]
@darrelmiller : FYI and in case you want to provide feedback

@schmid37, @gavinmnz, I think this should solve your issue. Please provide feedback (and try out the branch [jmprieur/SeparateGraph](https://github.com/AzureAD/microsoft-identity-web/tree/jmprieur/SeparateGraph)

File: src/Microsoft.Identity.Web.MicrosoftGraph/Microsoft.Identity.Web.MicrosoftGraph.csproj [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+    <DefineConstants>$(DefineConstants);WEB</DefineConstants>
+    <PackageRequireLicenseAcceptance>true</PackageRequireLicenseAcceptance>
+    <!-- Copyright needs to be in the form of © not (c) to be compliant -->
+    <Title>Microsoft Identity Web, Microsoft Graph helper</Title>
+    <Authors>Microsoft</Authors>
+    <Company>Microsoft Corporation</Company>
+    <Product>Microsoft Identity Web</Product>
+    <Description>
+      This package enables ASP.NET Core Web apps and Web APIs to use the Microsoft identity platform (formerly Azure AD v2.0).
+      This package is specifically used for web applications, which sign-in users, and protected web APIs, call Microsoft Graph.
```
Comment (pmaytak) [inline src/Microsoft.Identity.Web.MicrosoftGraph/Microsoft.Identity.Web.MicrosoftGraph.csproj:18]:
```suggestion
      This package is specifically used for web applications, which sign-in users, and protected web APIs, which call Microsoft Graph.
```

File: src/Microsoft.Identity.Web.MicrosoftGraph/Microsoft.Identity.Web.MicrosoftGraph.csproj [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+
+    <DefineConstants>$(DefineConstants);WEB</DefineConstants>
+    <PackageRequireLicenseAcceptance>true</PackageRequireLicenseAcceptance>
+    <!-- Copyright needs to be in the form of © not (c) to be compliant -->
+    <Title>Microsoft Identity Web, Microsoft Graph helper</Title>
+    <Authors>Microsoft</Authors>
+    <Company>Microsoft Corporation</Company>
+    <Product>Microsoft Identity Web</Product>
+    <Description>
+      This package enables ASP.NET Core Web apps and Web APIs to use the Microsoft identity platform (formerly Azure AD v2.0).
```
Comment (pmaytak) [inline src/Microsoft.Identity.Web.MicrosoftGraph/Microsoft.Identity.Web.MicrosoftGraph.csproj:17]:
```suggestion
      This package enables ASP.NET Core web apps and web APIs to use the Microsoft identity platform (formerly Azure AD v2.0).
```

File: src/Microsoft.Identity.Web.MicrosoftGraphBeta/Microsoft.Identity.Web.MicrosoftGraphBeta.csproj [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+
+    <DefineConstants>$(DefineConstants);WEB</DefineConstants>
+    <PackageRequireLicenseAcceptance>true</PackageRequireLicenseAcceptance>
+    <!-- Copyright needs to be in the form of © not (c) to be compliant -->
+    <Title>Microsoft Identity Web, Microsoft Graph helper</Title>
+    <Authors>Microsoft</Authors>
+    <Company>Microsoft Corporation</Company>
+    <Product>Microsoft Identity Web</Product>
+    <Description>
+      This package enables ASP.NET Core Web apps and Web APIs to use the Microsoft identity platform (formerly Azure AD v2.0).
```
Comment (pmaytak) [inline src/Microsoft.Identity.Web.MicrosoftGraphBeta/Microsoft.Identity.Web.MicrosoftGraphBeta.csproj:17]:
```suggestion
      This package enables ASP.NET Core web apps and web APIs to use the Microsoft identity platform (formerly Azure AD v2.0).
```

File: src/Microsoft.Identity.Web.MicrosoftGraphBeta/Microsoft.Identity.Web.MicrosoftGraphBeta.csproj [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+      This package is specifically used for web applications, which sign-in users, and protected web APIs, call Microsoft Graph Beta API.
```
Comment (pmaytak) [inline src/Microsoft.Identity.Web.MicrosoftGraphBeta/Microsoft.Identity.Web.MicrosoftGraphBeta.csproj:18]:
```suggestion
      This package is specifically used for web applications, which sign-in users, and protected web APIs, which call Microsoft Graph Beta API.
```

File: src/Microsoft.Identity.Web.MicrosoftGraph/Constants.cs [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+    internal static class Constants
+    {
+        public const string Authorization = "Authorization";
+        public const string Bearer = "Bearer";
+
+        // Microsoft Graph
+        public const string UserReadScope = "user.read";
+        public const string GraphBaseUrlV1 = "https://graph.microsoft.com/v1.0";
+    }
+}
```
Comment (jennyf19) [inline src/Microsoft.Identity.Web.MicrosoftGraph/Constants.cs:18]: hmmmm...not a fan of repeating code. especially for maintainability and fixes. i can look into how to resolve this.

**Comment by jennyf19** [comment]
looks great. one comment on the re-using of some of the constant classes. let's try to fix that before merging.

---
In reply to: [685816103](https://github.com/AzureAD/microsoft-identity-web/pull/534#issuecomment-685816103) [](ancestors = 685816103)

File: src/Microsoft.Identity.Web.MicrosoftGraph/Constants.cs (hunk 5 above)
Comment (pmaytak) [reply src/Microsoft.Identity.Web.MicrosoftGraph/Constants.cs:18]: Constants class in Identity.Web are internal. Perhaps set InterntalsVisibleTo to Identity.Web.Graph?

**Comment by jennyf19** [comment]
also...did you check the dependency across packages?

---
In reply to: [685852275](https://github.com/AzureAD/microsoft-identity-web/pull/534#issuecomment-685852275) [](ancestors = 685852275,685816103)

File: src/Microsoft.Identity.Web.MicrosoftGraph/Microsoft.Identity.Web.MicrosoftGraph.csproj (hunk 2 above)
Comment (jennyf19) [reply src/Microsoft.Identity.Web.MicrosoftGraph/Microsoft.Identity.Web.MicrosoftGraph.csproj:17]: @jmprieur i'll go through the files and fix this everywhere.

Comment (jennyf19) [reply src/Microsoft.Identity.Web.MicrosoftGraph/Microsoft.Identity.Web.MicrosoftGraph.csproj:17]: after you merge

File: src/Microsoft.Identity.Web.MicrosoftGraph/Constants.cs (hunk 5 above)
Comment (jmprieur) [reply src/Microsoft.Identity.Web.MicrosoftGraph/Constants.cs:18]: Thanks you for the PR comments. Addressed.

**Pull request merged**